overlord/devicestate/devicestate_remodel_test.go

```diff
@@ -5594,6 +5595,542 @@ plugs:
+		}},
+		Current:         snap.R(1),
+		Active:          true,
+		TrackingChannel: "latest/stable",
+	})
+
+	snapstate.Set(s.state, "pc-kernel", &snapstate.SnapState{
+		SnapType: "kernel",
+		Sequence: []*snap.SideInfo{{
+			RealName: "pc-kernel",
+			Revision: snap.R(1),
+			SnapID:   snaptest.AssertedSnapID("pc-kernel"),
+		}},
+		Current:         snap.R(1),
+		Active:          true,
+		TrackingChannel: "latest/stable",
+	})
+
+	snapstate.Set(s.state, "pc", &snapstate.SnapState{
+		SnapType: "gadget",
+		Sequence: []*snap.SideInfo{{
+			RealName: "pc",
+			Revision: snap.R(1),
+			SnapID:   snaptest.AssertedSnapID("pc"),
+		}},
+		Current:         snap.R(1),
+		Active:          true,
+		TrackingChannel: "latest/stable",
+	})
+
+	snapstate.Set(s.state, "snap-1", &snapstate.SnapState{
+		SnapType: "app",
+		Sequence: []*snap.SideInfo{{
+			RealName: "snap-1",
+			Revision: snap.R(1),
+			SnapID:   snaptest.AssertedSnapID("snap-1"),
+		}},
+		Current:         snap.R(1),
+		Active:          true,
+		TrackingChannel: "latest/stable",
+	})
+
+	validationSetInModel := map[string]interface{}{
+		"account-id": "canonical",
+		"name":       "vset-1",
+		"mode":       "enforce",
+	}
+
+	if sequence != "" {
+		validationSetInModel["sequence"] = sequence
+	}
+
+	newModel := s.brands.Model("canonical", "pc-model", map[string]interface{}{
+		"architecture":    "amd64",
+		"base":            "core20",
+		"revision":        "1",
+		"validation-sets": []interface{}{validationSetInModel},
+		"snaps": []interface{}{
+			map[string]interface{}{
+				"name":            "pc-kernel",
+				"id":              snaptest.AssertedSnapID("pc-kernel"),
+				"type":            "kernel",
+				"default-channel": "latest/stable",
+			},
+			map[string]interface{}{
+				"name":            "pc",
+				"id":              snaptest.AssertedSnapID("pc"),
+				"type":            "gadget",
+				"default-channel": "latest/stable",
+			},
+			map[string]interface{}{
+				"name":            "snap-1",
+				"id":              snaptest.AssertedSnapID("snap-1"),
+				"type":            "app",
+				"default-channel": "latest/stable",
+			},
+		},
+	})
+
+	vset, err := s.brands.Signing("canonical").Sign(asserts.ValidationSetType, map[string]interface{}{
+		"type":         "validation-set",
+		"authority-id": "canonical",
+		"series":       "16",
+		"account-id":   "canonical",
+		"name":         "vset-1",
+		"sequence":     "1",
+		"snaps": []interface{}{
+			map[string]interface{}{
+				"name":     "snap-1",
+				"id":       snaptest.AssertedSnapID("snap-1"),
+				"revision": "2",
+				"presence": "required",
+			},
+		},
+		"timestamp": time.Now().UTC().Format(time.RFC3339),
+	}, nil, "")
+	c.Assert(err, IsNil)
+
+	testDeviceCtx = &snapstatetest.TrivialDeviceContext{
+		Remodeling:     true,
+		DeviceModel:    newModel,
+		OldDeviceModel: currentModel,
+		CtxStore: &fakeSequenceStore{
+			fn: func(*asserts.AssertionType, []string, int, *auth.UserState) (asserts.Assertion, error) {
```

Comment:
shouldn't we check that key is the one we expect? same of other places using the fakeSequenceStore